pkg/ingester/incremental_transfer.go

```diff
@@ -0,0 +1,263 @@
+package ingester
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/ring"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/user"
+	"golang.org/x/net/context"
+)
+
+var (
+	blockedRanges = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
I'm happy to do this, but my understanding is that the goal was to move away from promauto, wasn't it?

Reply:
The goal is to move away from globally registered metrics. `promauto` is just syntactic sugar over what you're already doing. I haven't commented about passing the registry in input since this PR is already in a advanced state and didn't want to have you reconsider the design.

Reply:
Ah, understood. I'll change it to `promauto`. Thanks!